debugging/kernel-logger/kp_kernel_logger.cpp

```diff
@@ -80,19 +143,23 @@ extern "C" void kokkosp_begin_parallel_for(const char* name,
 }
 
 extern "C" void kokkosp_end_parallel_for(const uint64_t kID) {
+  int output = kID;
+  if (suppressCounts()) output = 0;
   printf("KokkosP: Execution of kernel %llu is completed.\n",
-         (unsigned long long)(kID));
+         (unsigned long long)output);
 }
 
 extern "C" void kokkosp_begin_parallel_scan(const char* name,
                                             const uint32_t devID,
                                             uint64_t* kID) {
-  *kID = uniqID++;
+  *kID       = uniqID++;
+  int output = *kID;
+  if (suppressCounts()) output = 0;
 
   printf(
-      "KokkosP: Executing parallel-scan kernel on device %d with unique "
+      "KokkosP: Executing parallel-scan kernel on device %s with unique "
       "execution identifier %llu\n",
-      devID, (unsigned long long)(*kID));
+      deviceIdToString(devID).c_str(), (unsigned long long)(output));
```

Comment:
Wait... What if we have several devices ? For instance, several `Cuda` GPUs.

Seems this change is really not going in the direction of allowing devices to be distinguished.

I would be OKish with:
```suggestion
      printf(
      "KokkosP: Executing parallel-scan kernel on device %d (%s) with unique "
      "execution identifier %llu\n",
      devID, deviceIdToString(devID).c_str(), (unsigned long long)(output));
```